File: manager-web/src/types/teacher.ts
export interface ClassSummary {
  id: string;
  name: string;
  school_name: string;
  class_code: string;
  student_count: number;
  slot_limit: number;
  status: string;
  expires_at: string | null;
}

export interface StudentSummary {
  user_id: string;
  full_name: string;
  avatar_url: string | null;
  progress_percent: number;
  current_lesson: string;
  avg_score: number | null;
  last_active: string | null;
  needs_support: boolean;
}

export interface LessonDetail {
  lesson_id: string;
  title: string;
  status: "completed" | "in_progress" | "locked" | "not_started" | string;
  score: number | null;
  time_spent_minutes: number | null;
  completed_at: string | null;
}

export interface StudentDetail {
  user_id: string;
  full_name: string;
  lessons: LessonDetail[];
}

export interface TeacherClassesResponse {
  classes: ClassSummary[];
}

export interface ClassStudentsResponse {
  class_id: string;
  class_name: string;
  students: StudentSummary[];
}

export interface ClassAssignmentsResponse {
  class_id: string;
  assignments: AssignmentSummary[];
}

export interface ImportedQuestionInput {
  question: string;
  options: string[];
  answer: number | string;
  explanation?: string;
}

export interface ImportedLessonCreate {
  title: string;
  description?: string;
  content?: string;
  questions: ImportedQuestionInput[];
}

export interface ImportedLessonSummary {
  id: string;
  class_id?: string;
  title: string;
  description: string | null;
  question_count?: number;
  quiz_count?: number;
  created_at: string | null;
}

export interface ClassLessonsResponse {
  class_id: string;
  lessons: ImportedLessonSummary[];
}

export interface QuizQuestionInput {
  id?: string | number;
  question: string;
  options: QuizOptionInput[];
  explanation?: string;
}

export interface QuizOptionInput {
  id: string;
  text: string;
  correct: boolean;
}

export interface AssignmentCreate {
  lesson_id: string;
  title?: string;
  due_at?: string | null;
}

export interface AssignmentSummary {
  id: string;
  lesson_id: string;
  class_id: string;
  title: string;
  lesson_title?: string | null;
  question_count?: number;
  assigned_count?: number;
  completed_count: number;
  total_students?: number;
  student_status?: "not_started" | "in_progress" | "completed" | null;
  student_score?: number | null;
  due_at: string | null;
  created_at?: string | null;
  assigned_at?: string | null;
  status?: string;
}

export interface AuthUser {
  id: string;
  email: string;
  full_name: string;
  role: string;
  avatar_url?: string | null;
}

export interface LoginResponse {
  access_token: string;
  refresh_token?: string;
  token_type?: string;
  user: AuthUser;
}

export interface AuthSession {
  token: string;
  user: AuthUser;
}

File: backend/models/teacher.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ClassSummary(BaseModel):
    id: str
    name: str
    school_name: str | None = None
    class_code: str
    student_count: int
    slot_limit: int
    status: str
    expires_at: datetime | None = None


class StudentSummary(BaseModel):
    user_id: str
    full_name: str
    avatar_url: str | None = None
    progress_percent: int
    current_lesson: str | None = None
    avg_score: float | None = None
    last_active: datetime | None = None
    needs_support: bool


class LessonDetail(BaseModel):
    lesson_id: str
    title: str
    status: str
    score: int | None = None
    time_spent_minutes: int | None = None
    completed_at: datetime | None = None


class StudentDetail(BaseModel):
    user_id: str
    full_name: str
    avatar_url: str | None = None
    class_id: str
    class_name: str
    progress_percent: int
    avg_score: float | None = None
    last_active: datetime | None = None
    needs_support: bool
    lessons: list[LessonDetail]


class TeacherClassesResponse(BaseModel):
    classes: list[ClassSummary]


class ClassStudentsResponse(BaseModel):
    class_id: str
    class_name: str
    students: list[StudentSummary]


class ImportedQuestion(BaseModel):
    question: str
    options: list[str]
    answer: int | str
    explanation: str | None = None


class ImportedLessonCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = None
    content: str | None = None
    questions: list[ImportedQuestion] = Field(default_factory=list)


class ImportedLessonSummary(BaseModel):
    id: str
    class_id: str
    title: str
    description: str | None = None
    question_count: int
    created_at: datetime | None = None


class AssignmentCreate(BaseModel):
    lesson_id: str
    title: str | None = None
    due_at: datetime | None = None


class AssignmentSummary(BaseModel):
    id: str
    class_id: str
    lesson_id: str
    title: str
    lesson_title: str | None = None
    question_count: int
    assigned_count: int = 0
    completed_count: int = 0
    student_status: Literal["not_started", "in_progress", "completed"] | None = None
    student_score: int | None = None
    submitted_at: datetime | None = None
    due_at: datetime | None = None
    created_at: datetime | None = None


class ClassLessonsResponse(BaseModel):
    lessons: list[ImportedLessonSummary]


class ClassAssignmentsResponse(BaseModel):
    assignments: list[AssignmentSummary]
